graylog2-server/src/main/java/org/graylog2/security/DefaultX509TrustManager.java

```diff
@@ -107,10 +119,12 @@ private void validateHostnames(X509Certificate[] x509Certificates, String s) thr
     }
 
     private boolean certificateMatchesHostname(X509Certificate x509Certificate) {
-        try {
-            return this.authorizer.verify(this.host, x509Certificate);
-        } catch (IOException e) {
-            return false;
-        }
+        return this.hosts.stream().anyMatch(host -> {
+            try {
+                return this.authorizer.verify(host, x509Certificate);
```

Comment:
Is it actually okay to just check if the certificate matches any host in our list? I was wondering if this can be an issue because we don't know which host is giving us the certificate. So a host could give us the certificate of another host from our list.

Reply:
That's difficult to implement. I've added a javadoc comment to point this out.
However, in our scenario, if I trust to talk to any of the servers in the list, it shouldn't matter which certificate they provide.

Reply:
Thanks! :+1: